### PR TITLE
Use common calico base and unify Dockerfiles

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

This change unifies Dockerfiles for all supported architectures into one and uses the common `calico/base` image. It also modifies Makefile to use host native calico/go-build for cross compiling.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,7 +36,7 @@ test: fmt vet ut
 # The target architecture is select by setting the ARCH variable.
 # When ARCH is undefined it is set to the detected host architecture.
 # When ARCH differs from the host architecture a crossbuild will be performed.
-ARCHES ?= $(patsubst build/Dockerfile.%,%,$(wildcard build/Dockerfile.*))
+ARCHES ?= amd64 arm64 ppc64le s390x
```
Comment (hjiawei) [inline Makefile:39]: We have a single Dockerfile now so the supported architectures are explicitly listed.

File: Makefile [hunk 2]
```diff
@@ -116,14 +92,14 @@ endif
 PACKAGE_NAME?=github.com/tigera/operator
 LOCAL_USER_ID?=$(shell id -u $$USER)
 GO_BUILD_VER?=v0.90
-CALICO_BUILD?=calico/go-build:$(GO_BUILD_VER)-$(ARCH)
+CALICO_BUILD?=calico/go-build:$(GO_BUILD_VER)-$(BUILDARCH)
```
Comment (hjiawei) [inline Makefile:95]: Use host native `calico/go-build` image for cross compiling all supported architectures.

File: Makefile [hunk 3]
```diff
@@ -89,15 +74,6 @@ DOCKER_CONFIG ?= $(HOME)/.docker/config.json
 escapefs = $(subst :,---,$(subst /,___,$(1)))
 unescapefs = $(subst ---,:,$(subst ___,/,$(1)))
 
-# Targets used when cross building.
-.PHONY: register
```
Comment (hjiawei) [inline Makefile:93]: The updated `Dockerfile` doesn't need any coreutils to run so qemu isn't needed.

File: build/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+LABEL description="Tigera Operator manages the lifecycle of a Calico or Calico Enterprise installation on Kubernetes or OpenShift"
+LABEL maintainer="maintainers@tigera.io"
+LABEL name="Tigera Operator"
+LABEL release="1"
+LABEL summary="Tigera Operator manages the lifecycle of a Calico or Calico Enterprise installation on Kubernetes or OpenShift"
+LABEL vendor="Tigera"
+LABEL version=${GIT_VERSION}
+
+COPY --from=source / /
```
Comment (hjiawei) [inline build/Dockerfile:40]: Squash into (1) calico/base and (2) operator specific layers.

File: Makefile [hunk 5]
```diff
@@ -150,6 +126,8 @@ CONTAINERIZED= mkdir -p .go-pkg-cache $(GOMOD_CACHE) && \
 		-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
 		-e GOPATH=/go \
 		-e GOCACHE=/go-cache \
+		-e GOOS=linux \
```
Comment (hjiawei) [inline Makefile:129]: operator runs on linux OS only.

File: build/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+USER 10001
```
Comment (rene-dekker) [inline build/Dockerfile:42]: Shouldn't this be 10001:10001?

Comment (hjiawei) [reply build/Dockerfile]: Updated.

**Review by rene-dekker — APPROVED** [review]
lgtm

**Pull request merged**